dashboard/modules/job/job_manager.py

```diff
@@ -315,8 +317,11 @@ async def _polling(self, child_process) -> int:
                     await asyncio.sleep(self.SUBPROCESS_POLL_PERIOD_S)
         except Exception:
             if child_process:
-                # TODO (jiaodong): Improve this with SIGTERM then SIGKILL
-                child_process.kill()
+                os.killpg(os.getpgid(child_process.pid), signal.SIGTERM)
```

Comment:
This might be overly defensive, but should we catch `ProcessLookupError` as in this PR? https://github.com/ray-project/ray/pull/30066

Reply:
Good idea, done!